src/DynamoCore/Logging/AnalyticsService.cs

```diff
@@ -17,8 +17,33 @@ class AnalyticsService
         /// the Analytics service and application life cycle start is tracked.
         /// </summary>
         /// <param name="model">DynamoModel</param>
-        internal static void Start(DynamoModel model)
+        /// <param name="disableADPForProcessLifetime">Pass true to disable ADP for the lifetime of the Dynamo or host process</param>
+        /// <param name="isHeadless">Analytics won't be started if IsHeadless, but ADP may be loaded to be disabled.</param>
+        /// <param name="isTestMode">Analytics won't be started if isTestMode, ADP will not be loaded.</param>
+        internal static void Start(DynamoModel model, bool disableADPForProcessLifetime, bool isHeadless, bool isTestMode)
         {
+            if (isTestMode)
+            {
+                return;
+            }
+            if (disableADPForProcessLifetime)
+            {
+                //if this returns false something has gone wrong.
+                //the client requested ADP be disabled, but we cannot disable it.
+                if (!ADPProcessSession.DisableADPForProcessLifetime())
+                {
+                    //TODO consider throwing instead - that will cause a crash.
+                    model.Logger.LogNotification("Dynamo",
+                        "Dynamo Startup Error",
+                        "Analytics could not be disabled",
+                        "Dynamo was started with configuration requesting ADP be disabled - but ADP could not be disabled.");
+                }
+            }
+
+            if (isHeadless)
```

Comment:
I modified how late we bail out if `isHeadless` is true because some clients need to disable ADP and still use Dynamo as a headless runner.

Reply:
Can you remind me, the `isHeadless` is set to true only when using command line, right? For Dynamo Player and GD, analytics is on because we are using Dynamo as a server.

Reply:
I believe player does not use headless, I am not sure about GD.